core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -247,14 +250,10 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeGenericValue(storedFieldNames);
```

Comment:
Can you add a note to this method about it being used to generate the cache key for the search request and having to be consistent? Hopefully we'll be more careful about it.